site/middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { NextResponse, type NextRequest } from 'next/server';
+
+export async function middleware(request: NextRequest) {
+  const country = request.geo.country || 'US'; // Default to 'US' if country is not detected
+
+  // List of blocked countries (ISO 3166-1 alpha-2 country codes)
+  const blockedCountries = ['US']; // Example: China, Russia, Iran
+
+  if (blockedCountries.includes(country)) {
+    return new NextResponse('Access Denied', { status: 403 });
+  }
+
+  return NextResponse.next();
+}
+
+export const config = {
+  matcher: '/swap/swap*', // Apply middleware to all paths
```

Comment:
Let's test this PR first with a testing page.

And then we use it for the entire site.

Reply:
Sounds good!

Reply:
It's been tested in Vercel Staging. Would you prefer to test a single page (/swap/swap) in prod as well before enabling for the entire website?